infra/src/test/kotlin/fi/fta/geoviite/infra/ui/pagemodel/map/TrackLayoutPage.kt

```diff
@@ -0,0 +1,137 @@
+        MM_50("50 mm"),
+        MM_100("100 mm"),
+        MM_200("200 mm"),
+        MM_500("500 mm"),
+        MM_1000("1000 mm"),
+        M_2("2 m"),
+        M_5("5 m"),
+        M_10("10 m"),
+        M_20("20 m"),
+        M_50("50 m"),
+        M_100("100 m"),
+        M_200("200 m"),
+        M_500("500 m"),
+        M_1000("1000 m"),
+        KM_2("2 km"),
+        KM_5("5 km"),
+        KM_10("10 km"),
+        KM_20("20 km"),
+        KM_50("50 km"),
+        KM_100("100 km"),
+        KM_200("200 km"),
+        KM_500("500 km")
+    }
+    
+    init {
+        mapScale
+    }
+
+    val toolPanel: E2EToolPanel by lazy { E2EToolPanel(this.elementFetch) }
+    val selectionPanel: E2ESelectionPanel by lazy { E2ESelectionPanel(this.elementFetch) }
+    val appBar: E2EAppBar by lazy { E2EAppBar() }
+    val toolBar: E2EToolBar by lazy { E2EToolBar(this.elementFetch) }
+    val mapScale: MapScale
+        get() {
+            val scale = childText(By.className("ol-scale-line-inner"))
+            return MapScale.values().first { it.value == scale }
+        }
+
+    companion object {
+        fun finishLoading() {
+            withNoImplicitWait { waitUntilDoesNotExist(By.className(".map__loading-spinner")) }
+        }
+    }
+
+    fun scrollMap(xOffset: Int, yOffset: Int): E2ETrackLayoutPage = apply {
+        logger.info("Scroll map x=$xOffset y=$yOffset")
+        val canvas = childElement(By.cssSelector("div.map"))
+
+        Actions(browser()).dragAndDropBy(canvas, xOffset, yOffset).build().perform()
+    }
+
+    fun clickAtCoordinates(point: Point, doubleClick: Boolean = false): E2ETrackLayoutPage = apply {
+        clickAtCoordinates(point.x, point.y, doubleClick)
+    }
+
+    fun clickAtCoordinates(point: LayoutPoint, doubleClick: Boolean = false): E2ETrackLayoutPage = apply {
+        clickAtCoordinates(point.x, point.y, doubleClick)
+    }
+
+    fun clickAtCoordinates(xPoint: Double, yPoint: Double, doubleClick: Boolean = false): E2ETrackLayoutPage = apply {
+        val pxlCoordinates =
+            javaScriptExecutor().executeScript("return map.getPixelFromCoordinate([$xPoint,$yPoint])").toString()
+                .replace("[^0-9.,]".toRegex(), "").split(",").map { doubleStr -> doubleStr.toDouble().roundToInt() }
+
+        logger.info("Map coordinates ($xPoint,$yPoint) are at $pxlCoordinates")
+        clickAtCoordinates(pixelX = pxlCoordinates[0], pixelY = pxlCoordinates[1], doubleClick)
+    }
+
+    fun clickAtCoordinates(pixelX: Int, pixelY: Int, doubleClick: Boolean = false): E2ETrackLayoutPage = apply {
+        logger.info("Click map at ($pixelX,$pixelY)")
+        val canvas = childElement(By.cssSelector("div.map"))
+        clickElementAtPoint(canvas, pixelX, pixelY, doubleClick)
+    }
+
+    fun switchToDraftMode(): E2ETrackLayoutPage = apply {
+        toolBar.switchToDraft()
+    }
+
+    fun goToPreview() = toolBar.goToPreview()
+
+    fun zoomToScale(targetScale: MapScale): E2ETrackLayoutPage = apply {
```

Comment:
One zoom to rule them all, Enumien ansioista kutsujan ei tarvitse enää tietää mihin suuntaa pitää zoomata. Aikaisemmin oli siis zoomOutToScale ja zoomInToScale